endpoints/v2/__init__.py

```diff
@@ -128,7 +128,10 @@ def wrapped(namespace_name, repo_name, *args, **kwargs):
             repository = namespace_name + "/" + repo_name
             if allow_public:
                 repository_ref = registry_model.lookup_repository(namespace_name, repo_name)
-                if repository_ref is None or not repository_ref.is_public:
+                if repository_ref is None:
+                    return func(namespace_name, repo_name, *args, **kwargs)
```

Comment:
This was working as intended. A request to a non-existing repo in a namespace that you don't have access to should not return 404. Otherwise it would be a leak about what exists or not in a namespace that you shouldn't have visibility into.

Reply:
We already discussed this on Slack - consistently returning a 404 doesn't leak anything, and it's in fact widely adopted api behaviour. Have a look at the thread for a refresher https://coreos.slack.com/archives/G7VFPAY7Q/p1640184124174900.

Reply:
Note that changing the behaviour is **not** the point of this PR though. Hopefully my other comments expand on that.